### ISSUE TITLE
Having trouble understanding how data is being initialized

### ISSUE THREAD

**Issue by madhavthaker** [llSourcell/wiki_generator_live#4]
We see that the data is being initialized here:

    data = tf.placeholder(tf.float32, [batch_size, len_per_section, char_size])

I'm having trouble understanding how this actually works. This is my understanding of the three parameters above. 

    batch_size = number of samples that is going into the model at a time
    len_per_section = number of characters we want to read at a time
    char_size = number of unique characters in our training set. 

Now, The data variable is referenced here:

    #unrolled LSTM loop
    #for each input set
    for i in range(len_per_section):
        #calculate state and output from LSTM
        output, state = lstm(data[:, i, :], output, state)
        #to start, 
        if i == 0:
            #store initial output and labels
            outputs_all_i = output
            labels_all_i = data[:, i+1, :]
        #for each new set, concat outputs and labels
        elif i != len_per_section - 1:
            #concatenates (combines) vectors along a dimension axis, not multiply
            outputs_all_i = tf.concat(0, [outputs_all_i, output])
            labels_all_i = tf.concat(0, [labels_all_i, data[:, i+1, :]])
        else:
            #final store
            outputs_all_i = tf.concat(0, [outputs_all_i, output])
            labels_all_i = tf.concat(0, [labels_all_i, labels])

This is where I'm a bit confused. We are iterating through each `len_per_section` so does that mean we are providing 50 characters in each loop as we go through the batch? Even then, I don't know where the third parameter `char_size` comes into play.
